skbio/io/format/gff3.py

```diff
@@ -0,0 +1,508 @@
+        # the 1st column is seq ID for every feature. don't store
+        # this repetitive information
+        metadata = {'source': columns[1],
+                    'type': columns[2],
+                    'score': columns[5],
+                    'strand': columns[6]}
+        phase = columns[7]
+        # phase value can only be int or '.'
+        try:
+            metadata['phase'] = int(phase)
+        except ValueError:
+            if phase != '.':
+                raise GFF3FormatError(
+                    'unknown value for phase column: {!r}'.format(phase))
+        metadata.update(_parse_attr(columns[8]))
+
+        start, end = columns[3:5]
+
+        bounds = [(int(start)-1, int(end))]
+
+        interval_metadata.add(bounds, metadata=metadata)
+
+    return interval_metadata
+
+
+def _parse_attr(s):
+    '''parse attribute column'''
+    voca_change = _vocabulary_change('gff3')
+    md = {}
+    # in case the line ending with ';', strip it.
+    s = s.rstrip(';')
+    for attr in s.split(';'):
+        k, v = attr.split('=')
+        if k in voca_change:
+            k = voca_change[k]
+        md[k] = v
+    return md
+
+
+def _serialize_interval_metadata(
+        interval_metadata, seq_id, fh, skip_subregion=True):
+    '''Serialize an IntervalMetadata to GFF3.
+
+    Parameters
+    ----------
+    interval_metadata : IntervalMetadata
+    skip_subregion : bool
+        whether to skip outputting each sub region as a line in GFF3.
+    '''
+    # write file header
+    fh.write('##gff-version 3\n')
+
+    column_keys = ['source', 'type', 'score', 'strand', 'phase']
+    voca_change = _vocabulary_change('gff3', False)
+    voca_skip = _vocabulary_skip('gff3')
+    voca_skip.extend(column_keys)
+
+    # these characters have reserved meanings in column 9 and must be
+    # escaped when used in other contexts
+    escape = str.maketrans({';': '%3B',
+                            '=': '%3D',
+                            '&': '%26',
+                            ',': '%2C'})
+
+    for interval in interval_metadata._intervals:
+        md = interval.metadata
+        bd = interval.bounds
+        start = str(bd[0][0] + 1)
+        end = str(bd[-1][-1])
+
+        source, feat_type, score, strand, phase = [
+            str(md.get(i, '.')) for i in column_keys]
+        columns = [seq_id, source, feat_type, start, end, score, strand, phase]
+
+        # serialize the attributes in column 9
+        attr = []
+        # use sort to make the output order deterministic
+        for k in sorted(md):
+            if k in voca_skip:
+                # skip the metadata don't go to attribute column
+                continue
+            if k in voca_change:
+                k = voca_change[k]
+            v = md[k]
+            attr.append('%s=%s' % (k.translate(escape), v.translate(escape)))
+        columns.append(';'.join(attr))
+
+        print('\t'.join(columns), file=fh)
```

Comment:
Didn't change that because I need another line to add a linefeed if using `fh.write()`. Thought there is no preference using `write` or `print` to output to file. Can change that if `write` is appealing more to you.